### PR TITLE
Username enumeration in dolibarr

### PR DESCRIPTION
**Pull request by kajalNair**
## ✍️ Description

The application is vulnerable to username enumeration in password reset functionality.

## 🕵️‍♂️ Proof of Concept

```
// Validate new password
if ($action == 'validatenewpassword' && $username && $passwordhash)
{
    $edituser = new User($db);
    $result = $edituser->fetch('', $_GET["username"]);
    if ($result < 0)
    {
        $message = '<div class="error">'.$langs->trans("ErrorLoginDoesNotExists", $username).'</div>';
    } else {
        if (dol_verifyHash($edituser->pass_temp, $passwordhash))
```

## 💥 Impact

The attacker would be able to brute force the user accounts.

## ☎️ Contact

No

## ✅ Checklist

**In my pull request, I have:**

- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

File: bounties/packagist/dolibarr/dolibarr/4/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  "Package": {
+    "Registry": "packagist",
+    "Name": "dolibarr/dolibarr",
+    "URL": "https://packagist.org/packages/dolibarr/dolibarr"
+  },
+  "CWEs": [
+    {
+      "ID": "200",
+      "Description": "Exposure of Sensitive Information to an Unauthorized Actor
+"
```
Comment (JamieSlome) [inline bounties/packagist/dolibarr/dolibarr/4/vulnerability.json:19]: Just pop the speech marks back up to the previous line! 🍰

Comment (kajalNair) [reply bounties/packagist/dolibarr/dolibarr/4/vulnerability.json]: done.
This has been already reviewed by @Mik317. A new PR is created against the closed one.
Thanks! 👍

**Review by Mik317 — COMMENTED** [review]
Hi @kajalNair :smile:,
I'm not sure if we can consider this issue to be valid since `dolibarr` provides a `rate limit` defense mechanism which allows managers/admins to avoid `bruteforcing` attacks.

@bbeale @mufeedvh @418raj what do you think about it?

Cheers,
Mik

**Comment by bbeale** [comment]
Actually, it looks like this one is fixed in the dev branch. I'll edit this comment later with a link to relevant commits, but if you build dolibarr from the code in the dev branch, the vuln is no longer there and replaced with a more appropriately vague user message: 

dolibarr-poc-3

**Comment by 418raj** [comment]
Thanks for the disclosure! Unfortunately since it's been fixed on their dev branch we can't award this one. Cheers 👍

**Pull request closed without merging**